Gate_system/Models/ReportItem.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Gate_system.Models
+{
+    public class ReportItem
+    {
+
+        private int id;
+        private string firstName;
+        private string lastName;
+        private DateTime date;
+        private GateName gateTried;
+        private Boolean wasGateEntered;
+        private GateFunction whatWasDone;
+
+        public int Id { get { return id; } }
+
+        public string FirstName { get { return firstName;} }
+
+        public string LastName { get { return lastName;} }
+
+        public DateTime Date { get { return date; } }
+
+        public GateName GateTried { get { return gateTried; } }
+
+        public Boolean WasGateEntered { get { return wasGateEntered; } }
+
+        public GateFunction WhatWasDone { get { return whatWasDone; } }
+
+        public ReportItem(int id, string firstName, string lastName, DateTime date, GateName gateTried, Boolean wasGateEntered, GateFunction whatWasDone)
+        {
+            this.id = id;
```

Comment:
Nors matau, kad tai veikia, bet C# nera iprasta naudoti this, nebent su extension'ais. Zymiai dazniau pasitaikantis variantas - duoti skirtingus pavadinimus parametrams ir vidinems properties, prideti koki nors bruksneli, mazaja/didziaja raide ir pan.